### ISSUE TITLE
[BUG] Impossible to use DQL Search query with $ inside

### ISSUE THREAD

**Issue by markusg80** [opensearch-project/OpenSearch-Dashboards#5630]
**Describe the bug**

**To Reproduce**
1. Go to Discover
2. Enter Search String like "winlog.event_data.TargetUserName: *$ 

**Expected behavior**
Documents with parameter winlog.event_data.TargetUserName and value ending with $ (Computer Accouts) are filtered out 

**OpenSearch Version**
2.11.1

**Dashboards Version**
2.11.1

**Screenshots**
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/12497477/ed0940b5-9713-4577-9a4b-216063077f1e)

**Host/Environment (please complete the following information):**
 - OS: Rocky linux 9.3
 - Firefox 120.0.1

I also cannot escape the $

**Comment by ashwin-pc** [comment]
The DQL query under the hood is converted into a DSL request like below. Are you able to able to query using DSL the fields with `$` in them?

```json
{
  "query": {
    "bool": {
      "must": [],
      "filter": [
        {
          "multi_match": {
            "type": "best_fields",
            "query": "$",
            "lenient": true
          }
        }
      ]
    }
  }
}
```

**Comment by markusg80** [comment]
I try also with this. But it not works.

Should this line
"query": "$"

Not be 
"query": "*$"

Von meinem iPhone gesendet

> Am 29.12.2023 um 00:07 schrieb Ashwin P Chandran ***@***.***>:
> 
> "query": "$"

**Comment by markusg80** [comment]
Is there any further information i have to provide? With the proposed solution of @ashwin-pc i don't get any results.

**Comment by ashwin-pc** [comment]
Nope, thi was useful @markusg80. Adding more context here. To reproduce the issue with DQL search queries involving the $ character, follow these steps:

Open the Dashboards Console app and execute the following commands to create an index and ingest data:

```json
PUT /test_index
{
  "mappings": {
    "properties": {
      "winlog.event_data.TargetUserName": {
        "type": "keyword"
      }
    }
  }
}

POST /test_index/_bulk
{ "index": {} }
{ "winlog.event_data.TargetUserName": "user1$" }
{ "index": {} }
{ "winlog.event_data.TargetUserName": "user2" }
{ "index": {} }
{ "winlog.event_data.TargetUserName": "admin$" }
```

Go to the "Discover" tab and select the test_index index pattern.

Enter winlog.event_data.TargetUserName: *$ in the search bar and observe the results.

This should help reproduce the issue described.

**Comment by markusg80** [comment]
With this test i get back two lines

**Comment by markusg80** [comment]
![Image](https://github.com/user-attachments/assets/4bdb00f9-7b51-4852-8d63-4f4dc68eec1d)

**Comment by markusg80** [comment]
this is what i get back

**Comment by markusg80** [comment]
What is now the next steps?

**Comment by ashwin-pc** [comment]
So you are able to search with $ in te query bar. So it looks like there is some other issue here. Whats your original index mapping for the problematic index pattern?
